docs/utils/slurm.md

```diff
@@ -1,47 +1,155 @@
 # Parameter space discovery via slurm
 
-This module allows submitting a slurm job array where each
-individual job runs a simulation with a different set of parameters.
+This module allows submitting a python callback as a slurm job array where each
+individual job runs the callback with a different set of parameters.
 
-For a given problem `ExampleProblem` with design type `ExampleDesign`,
-the following code will submit a slurm job array with 3 jobs:
+```{testsetup} slurm
+import numpy as np
+from engibench.utils import slurm
+import fake_sbatch
+from examples.beams2d_patched import run_job
 
-```py
-static_args = slurm.Args(simulate_args={"config": {"sim_arg": 10}}, problem_args={"some_arg": True})
+slurm_args = slurm.SlurmConfig(sbatch_executable=fake_sbatch.__file__)
+np.set_printoptions(edgeitems=1)
+```
+
+**Step 1:** Define a callback and the parameter space
+
+```{literalinclude} examples/beams2d.py
+```
+
+```{testcode} slurm
 parameter_space = [
-     slurm.Args(problem_args={"problem_arg": 1}, design_args={"design_arg": -1}),
-     slurm.Args(problem_args={"problem_arg": 2}, design_args={"design_arg": -2}),
-     slurm.Args(problem_args={"problem_arg": 3}, design_args={"design_arg": -3}),
+    {
+        "config": {
+            "volfrac": volfrac,
+            "forcedist": forcedist,
+        }
+    }
+    for forcedist in [0.0, 0.5]
+    for volfrac in [0.1, 0.25, 0.8]
 ]
-slurm.submit(
-    problem=ExampleProblem,
-    static_args=static_args,
-    parameter_space=parameter_space),
+```
+
+**Step 2:** Specify arguments to be passed to slurm:
+
+```{testcode} slurm-real
+from engibench.utils import slurm
+slurm_args = slurm.SlurmConfig(runtime="1:00:00")
+```
+
+**Step 3:** Submit a slurm job array
+
+```{testcode} slurm
+
+job = slurm.sbatch_map(
+    run_job,
+    args=parameter_space,
+    slurm_args=slurm_args,
 )
 ```
 
-For example the first job would execute:
+For example this would submit a slurm array with 6 elements running
 
 ```py
-problem = ExampleProblem(some_arg=True, problem_arg=1)
-design = ExampleDesign(design_arg=-1)
-problem.simulate(design, config={"sim_arg": 10})
+run_job(config={"volfrac": 0.1}, "forcedist": 0.0) # element 1
```

Comment:
I think `forcedist` should be in the dict

Reply:
Thanks.